sklearn/tests/test_impute.py

```diff
@@ -210,6 +231,161 @@ def test_imputation_most_frequent():
+    # Test imputation using the constant strategy on integers
+    X = np.array([
+        [-1, 2, 3, -1],
+        [4, -1, 5, -1],
+        [6, 7, -1, -1],
+        [8, 9, 0, -1]
+    ])
+
+    X_true = np.array([
+        [0, 2, 3, 0],
+        [4, 0, 5, 0],
+        [6, 7, 0, 0],
+        [8, 9, 0, 0]
+    ])
+
+    imputer = SimpleImputer(missing_values=-1, strategy="constant",
+                            fill_value=0)
+    X_trans = imputer.fit_transform(X)
+
+    assert_array_equal(X_trans, X_true)
+
+
+@pytest.mark.parametrize("format", ["csr", "array"])
+def test_imputation_constant_float(format):
+    # Test imputation using the constant strategy on floats
+    X = np.array([
+        [np.nan, 1.1, 2.2, np.nan],
+        [3.3, np.nan, 4.4, np.nan],
+        [5.5, 6.6, np.nan, np.nan],
+        [7.7, 8.8, 9.9, np.nan]
+    ])
+
+    X = sparse.csr_matrix(X) if format == "csr" else X
+
+    X_true = np.array([
+        [0, 1.1, 2.2, 0],
+        [3.3, 0, 4.4, 0],
+        [5.5, 6.6, 0, 0],
+        [7.7, 8.8, 9.9, 0]
+    ])
+
+    if format == "csr":
+        X_true = sparse.csr_matrix(X_true)
+        X_true[np.array([[True, False, False, True],
+                         [False, True, False, True],
+                         [False, False, True, True],
+                         [False, False, False, True]])] = 0
+
+    imputer = SimpleImputer(strategy="constant", fill_value=0)
+    X_trans = imputer.fit_transform(X)
+
+    assert_allclose_dense_sparse(X_trans, X_true)
+
+
+@pytest.mark.parametrize("marker", [None, np.nan, "NAN", "", 0])
+def test_imputation_constant_object(marker):
+    # Test imputation using the constant strategy on objects
+    X = np.array([
+        [marker, "a", "b", marker],
+        ["c", marker, "d", marker],
+        ["e", "f", marker, marker],
+        ["g", "h", "i", marker]
+    ], dtype=object)
+
+    X_true = np.array([
+        ["missing", "a", "b", "missing"],
+        ["c", "missing", "d", "missing"],
+        ["e", "f", "missing", "missing"],
+        ["g", "h", "i", "missing"]
+    ], dtype=object)
+
+    imputer = SimpleImputer(missing_values=marker, strategy="constant",
+                            fill_value="missing")
+    X_trans = imputer.fit_transform(X)
+
+    assert_array_equal(X_trans, X_true)
+
+
+@pytest.mark.parametrize("dtype", [object, "category"])
+def test_imputation_constant_pandas(dtype):
+    # Test imputation using the constant strategy on pandas df
+    pd = pytest.importorskip("pandas")
+
+    f = io.StringIO(u"Cat1,Cat2,Cat3,Cat4\n"
+                    ",i,x,\n"
+                    "a,,y,\n"
+                    "a,j,,\n"
+                    "b,j,x,")
+
+    df = pd.read_csv(f, dtype=dtype)
+
+    X_true = np.array([
+        ["missing", "i", "x", "missing"],
+        ["a", "missing", "y", "missing"],
+        ["a", "j", "missing", "missing"],
+        ["b", "j", "x", "missing"]
+    ], dtype=object)
+
+    imputer = SimpleImputer(strategy="constant", fill_value="missing")
```

Comment:
I would do one of those tests without the specifying the `fill_value`, so we also test the actual default of "missing_value"